Client/Frontend/Browser/MetadataParserHelper.swift

```diff
@@ -46,5 +48,13 @@ class MetadataParserHelper: TabEventHandler {
             ]
             NotificationCenter.default.post(name: .OnPageMetadataFetched, object: nil, userInfo: userInfo)
         }
+        
+        // Ecosia: read local storage value for cookie consent
+        if url.normalizedHost?.hasPrefix("ecosia.org") == true, !tab.isPrivate {
+            webView.evaluateJavascriptInDefaultContentWorld("localStorage.getItem(\"\(Cookie.consentKey)\")") { result, error in
+                guard error == nil, let consentValue = result as? String else { return }
+                User.shared.cookieConsentValue = consentValue
+            }
+        }
```

Comment:
I have a suggestion to standardise cookie handling in our application:

We use `localStorage` at the moment, which works as a fallback (see [documentation](https://ecosia.atlassian.net/wiki/spaces/DEV/pages/4128796/Cookies#ECCC) under `Can it move to Local Storage?`), but is not actually the cookie. How about we move this inside our existing [Cookie](https://github.com/ecosia/ios-core/blob/main/Sources/Cookie.swift) class since it already reads exactly this cookie and could then save the value inside `User`?

As with regards to setting the cookie back, (done below in `UserScriptManager`), it could work exactly [how we do for the main cookie already](https://github.com/ecosia/ios-browser/blob/main/Client/Frontend/Browser/TabManager.swift#L113-L114) and also be standardised as a common behaviour of all cases inside `enum Cookie`.

I understand this is copied from **production code already**, so definitely not a blocker and would be even better to handle as a different ticket to properly QA and verify it works the same, but I believe is a nice way to make the code cleaner and more maintainable.

Reply:
We can probably get @ecotopian's input here in case I'm missing some context on why it has to use `localStorage`, but otherwise and if you both agree I can create a follow up ticket for it 🙂

Reply:
Thanks for the suggestion @lucaschifino ! 
Far from having strong opinions on that. 
I implemented it so to minimize the codebase differences with production.
Happy to take the occasion and further improve it as part of this work already!